src/main/java/io/vertx/core/http/impl/Http1xUpgradeToH2CHandler.java

```diff
@@ -26,10 +27,12 @@
   private final HttpServerChannelInitializer initializer;
   private final HandlerHolder<? extends Handler<HttpServerConnection>> holder;
   private VertxHttp2ConnectionHandler<Http2ServerConnection> handler;
+  private HttpServerOptions httpServerOptions;
```

Comment:
rather use 2 booleans for compression/decompression supported than the full options

Reply:
fixed